src/procedures/__tests__/ToggleAllowBeneficialInvestments.ts

```diff
@@ -0,0 +1,263 @@
+    });
+  });
+
+  describe('ToggleAllowBeneficialInvestments', () => {
+    test('should add the transaction to the queue to toggle allowed beneficial investments in a simple sto', async () => {
+      const addTransactionSpy = spy(target, 'addTransaction');
+      simpleStoMock.mock(
+        'changeAllowBeneficialInvestments',
+        Promise.resolve('ChangeAllowBeneficialInvestments')
+      );
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(simpleStoMock.getMockInstance().changeAllowBeneficialInvestments)
+      ).toEqual(true);
+      expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(
+        PolyTransactionTag.ChangeAllowBeneficialInvestments
+      );
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should add the transaction to the queue to toggle beneficial investments in a tiered sto', async () => {
+      moduleWrapperFactoryMock.mock('getModuleInstance', tieredStoMock.getMockInstance());
+      target = new ToggleAllowBeneficialInvestments(tieredParams, contextMock.getMockInstance());
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+      tieredStoMock.mock(
+        'changeAllowBeneficialInvestments',
+        Promise.resolve('ChangeAllowBeneficialInvestments')
+      );
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(tieredStoMock.getMockInstance().changeAllowBeneficialInvestments)
+      ).toEqual(true);
+      expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(
+        PolyTransactionTag.ChangeAllowBeneficialInvestments
+      );
+      expect(addTransactionSpy.callCount).toEqual(1);
+    });
+
+    test('should throw if there is an invalid sto address', async () => {
+      target = new ToggleAllowBeneficialInvestments(
+        {
+          ...simpleParams,
+          stoAddress: 'invalid',
+        },
+        contextMock.getMockInstance()
+      );
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.InvalidAddress,
+          message: `Invalid STO address invalid`,
+        })
+      );
+    });
+
+    test('should throw if trying to disallow beneficial investments', async () => {
+      target = new ToggleAllowBeneficialInvestments(
+        {
+          ...simpleParams,
+          allowBeneficialInvestments: false,
+        },
+        contextMock.getMockInstance()
+      );
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `Beneficial investments are already disallowed`,
+        })
+      );
+    });
+
+    test('should throw if beneficial investments are already allowed', async () => {
```

Comment:
"should throw if trying to allow beneficial investments when they are already allowed"